src/Box.cpp

```diff
@@ -7,46 +7,13 @@ namespace geo {
 Box::Box(const Vector3 &min, const Vector3 &max) {
     bounds[0] = min;
     bounds[1] = max;
-
-    int p0 = mesh_.addPoint(min.x, min.y, min.z); // 0
-    int p1 = mesh_.addPoint(max.x, min.y, min.z); // 1
-    int p2 = mesh_.addPoint(min.x, max.y, min.z); // 2
-    int p3 = mesh_.addPoint(max.x, max.y, min.z); // 3
-    int p4 = mesh_.addPoint(min.x, min.y, max.z); // 4
-    int p5 = mesh_.addPoint(max.x, min.y, max.z); // 5
-    int p6 = mesh_.addPoint(min.x, max.y, max.z); // 6
-    int p7 = mesh_.addPoint(max.x, max.y, max.z); // 7
-
-    // back plane
-    mesh_.addTriangle(p1, p0, p2);
-    mesh_.addTriangle(p1, p2, p3);
-
-    // front plane
-    mesh_.addTriangle(p4, p5, p6);
-    mesh_.addTriangle(p6, p5, p7);
-
-    // left plane
-    mesh_.addTriangle(p0, p4, p2);
-    mesh_.addTriangle(p2, p4, p6);
-
-    // right plane
-    mesh_.addTriangle(p5, p1, p3);
-    mesh_.addTriangle(p5, p3, p7);
-
-    // top plane
-    mesh_.addTriangle(p0, p1, p4);
-    mesh_.addTriangle(p4, p1, p5);
-
-    // bottom plane
-    mesh_.addTriangle(p3, p2, p6);
-    mesh_.addTriangle(p3, p6, p7);
```

Comment:
The mesh is now generated, when the mesh is requested. I did this, because the box can be changed overtime. In that case the mesh becomes invalid. There was no way generate the mesh. So you were able to `break` you box. In the new situation the mesh is cleared in case the size of the box in changed.

Reply:
What I am worried about is a function in Shape.cpp using mesh_ in a function. That would mean that that function would not work for Boxes. I would then either update the mesh when the shape is updated, maybe separate the mesh creation function. Or I would question why the box can be changed over time. 
This only seems to happen in the enclose function and I would argue that that should instead generate a new box object